### PR TITLE
Add type hints to group.py

### PR DESCRIPTION
**Pull request by KoStehner**
Hello,

I added annotations to `tables/group.py`.

Best regards
Ko

### CODE REVIEW

File: tables/group.py [hunk 1]
```diff
@@ -693,7 +709,7 @@ def _f_list_nodes(self, classname=None):
 
         return list(self._f_iter_nodes(classname))
 
-    def _f_iter_nodes(self, classname=None):
+    def _f_iter_nodes(self, classname: Optional[str]=None) -> Generator[Node, None, None]:
```
Comment (avalentino) [inline tables/group.py:712]: See comment above

**Review by avalentino — CHANGES_REQUESTED** [review]

File: tables/group.py [hunk 2]
```diff
@@ -421,7 +432,7 @@ def __getitem__(self, childname):
         except NoSuchNodeError:
             raise IndexError(childname)
 
-    def _f_walknodes(self, classname=None):
+    def _f_walknodes(self, classname: Optional[str]=None) -> Generator[Node, None, None]:
```
[the remark this answers is not in the document]
Comment (KoStehner) [reply tables/group.py]: Switched from `Generator` to `Iterator`. Thanks!

File: tables/group.py (hunk 1 above)
Comment (KoStehner) [reply tables/group.py]: Switched from `Generator` to `Iterator`

File: tables/group.py [hunk 3]
```diff
@@ -732,7 +748,7 @@ def _f_iter_nodes(self, classname=None):
                 if isinstance(childnode, class_):
                     yield childnode
 
-    def _f_walk_groups(self):
+    def _f_walk_groups(self) -> Generator["Group", None, None]:
```
[the remark this answers is not in the document]
Comment (KoStehner) [reply tables/group.py]: Switched from `Generator` to `Iterator`

**Review by avalentino — APPROVED** [review]

**Comment by avalentino** [comment]
Thanks @KoStehner

**Pull request merged**